### DISCUSSION TITLE
Is it possible to run a cluster of shlink servers behind a load balancer, with shlink-web-client and a single database?

### DISCUSSION THREAD

**Discussion by edv-wieland** [shlinkio/discussions#153] [Help wanted] [answered]
### Shlink version

4.2.3

### PHP version

8.4.1

### How do you serve Shlink

Self-hosted nginx

### Database engine

MariaDB

### Database version

10.5.26

### Summary

Hi,
I wanted to check if it is possible to run shlink on multiple servers with one backend and behind a load balancer?
I want to set up two or more instances of shlink with shlink-web-client and run them behind a hapee load balancer and a mssql database. Is this possible? What else do I need to be able to update each instance without breaking the database?

**Comment by acelaya** [comment]
When you say "one backend" I guess you mean one database server. 

This is definitely possible. You can have multiple Shlink instances all connected to the same database server. Just make sure they use different databases. 

You would only need one shlink-web-client instance also, as it can handle multiple servers at once, even on different versions of Shlink.

**Reply by edv-wieland** [reply]
Thanks for the fast reply.

So how can i make sure, that all shlink servers serve the same links? I hoped to achive this with one single mssql database that all instances can connect to. Because when one server fails I want the other shlink servers to still serve the links and also to have the web-client present.

**Reply by acelaya** [reply, marked as answer]
Sorry, I misunderstood.

What you want is to setup a cluster of Shlink instances. That's also possible, and in that case you should point them to the same database.

When you need to update Shlink, simply downscale to one instance, update it, and then upscale again to the desired amount of instances, but this process would depend on how you orchestrate and automate your infrastructure.

**Reply by acelaya** [reply]
This might be worth checking https://shlink.io/documentation/install-docker-image/#multi-instance-considerations

(that piece of documentation should not be in the docker page. I'll probably move it somewhere else)

**Reply by edv-wieland** [reply]
But there won't be any problem with the single database in the shlink cluster when upgrade all instances one by one?

**Reply by acelaya** [reply]
There should be no problem, but as I mentioned, I would recommend downscaling to one instance before updating to be double safe.

**Reply by edv-wieland** [reply]
I will try when there are updates to be installed
